x-team/xp-ui
# Fix high and critical vulnerabilities

**Release Type:** *Dev Improvement*

Fixes https://github.com/x-team/xp/issues/2046 and https://github.com/x-team/xp/issues/1974

## Description

It is fixing the critical vulnerabilities and most of the highs, the only high vulnerability left is related to the `parsejson` package that doesn't have a patch to fix it yet.

## Checklist

**Before submitting a pull request,** please make sure the following is done:



## Steps to Test or Reproduce

In `xp-ui`:

1. Run `rm -rf node_modules/ && npm i`
2. Run `npm audit`
3. Observe that the report of vulnerabilities has none marked as critical and just 1 as high

## Screenshots

- Current status of the vulnerabilities

![Screen Shot on 2020-04-10 at 02:10:57](https://user-images.githubusercontent.com/6010411/78967467-8e746380-7ad0-11ea-9773-ba06ef0f13c7.png)

- The only `parsejson` package that can't be fixed yet

![Screen Shot 2020-03-25 at 02 03 45](https://user-images.githubusercontent.com/6010411/77507550-068b2a00-6e3f-11ea-873b-0d94ef4bd7af.png)

- How is this high vulnerability affecting the project? 

The article published here said https://npmjs.com/advisories/528

> Affected versions of parsejson are vulnerable to a regular expression denial of service when parsing untrusted user input.
